regexp_MaintainCaptureOrder_test.go

```diff
@@ -0,0 +1,394 @@
+package regexp2
+
+import (
+	"testing"
+)
+
+func TestMaintainCaptureOrder_Basic(t *testing.T) {
+	r, err := Compile("(?<first>this).+?(testing).+?(?<last>stuff)", MaintainCaptureOrder)
```

Comment:
can we get a sister test where `MaintainCaptureOrder` isn't set?  That'd let us confirm the old behavior of `Groups()` is unaffected.